dev-tools/xtask/src/check_features.rs

```diff
@@ -0,0 +1,208 @@
+    /// version.
+    #[clap(long, value_name = "VERSION")]
+    install_version: Option<String>,
+}
+
+/// Run `cargo hack check`.
+pub fn run_cmd(args: Args) -> Result<()> {
+    // We cannot specify both `--ci` and `--install-version`, as the former
+    // implies we are using a pre-built version.
+    if args.ci && args.install_version.is_some() {
+        bail!("cannot specify --ci and --install-version together");
+    }
+
+    let cargo =
+        std::env::var("CARGO").unwrap_or_else(|_| String::from("cargo"));
+
+    let mut command = Command::new(&cargo);
+
+    // Add the `hack check` subcommand.
+    command.args(&["hack", "check"]);
+
+    if args.ci {
+        install_prebuilt_cargo_hack(&cargo)?;
+
+        let ex = if let Some(mut features) = args.exclude_features {
+            // Extend the list of features to exclude with the CI defaults.
+            features.extend(
+                CI_EXCLUDED_FEATURES.into_iter().map(|s| s.to_string()),
+            );
+
+            // Remove duplicates.
+            let excludes = features.into_iter().collect::<HashSet<_>>();
+
+            excludes.into_iter().collect::<Vec<_>>().join(",")
+        } else {
+            CI_EXCLUDED_FEATURES.join(",")
+        };
+
+        // Add the `--exclude-features` flag if we are running in CI mode.
+        command.args(["--exclude-features", &ex]);
+    } else {
+        install_cargo_hack(&cargo, args.install_version)?;
+        // Add "only" the `--exclude-features` flag if it was provided.
+        if let Some(features) = args.exclude_features {
+            command.args(["--exclude-features", &features.join(",")]);
+        }
+    }
+
+    if let Some(depth) = args.depth {
+        command.args(&["--depth", &depth.to_string()]);
+    }
+
+    // Pass along the `--message-format` flag if it was provided.
+    if let Some(fmt) = args.message_format {
+        command.args(["--message-format", &fmt]);
+    }
+
+    command
+        // Make sure we check everything.
+        .arg("--workspace")
+        // We want to check the binaries.
+        .arg("--bins")
+        // We want to check the feature powerset.
+        .arg("--feature-powerset")
+        // We will not check the dev-dependencies, which should covered by tests.
+        .arg("--no-dev-deps");
+
+    exec(command)
+}
+
+/// The supported operating systems.
+enum Os {
+    Illumos,
+    Linux,
+    Mac,
+}
+
+/// Get the current OS.
+fn os_name() -> Result<Os> {
+    let os = match std::env::consts::OS {
+        "linux" => Os::Linux,
+        "macos" => Os::Mac,
+        "solaris" | "illumos" => Os::Illumos,
+        other => bail!("OS not supported: {other}"),
+    };
+    Ok(os)
+}
+
+/// Get the path to the `out` directory.
+fn out_dir() -> Utf8PathBuf {
+    if let Ok(omicron_dir) = std::env::var("OMICRON") {
+        Utf8Path::new(format!("{}/out/cargo-hack", omicron_dir).as_str())
+            .to_path_buf()
```

Comment:
This can just be `format!("{}/out/cargo-hack").into()`

Reply:
derp yep.